src/Compilers/CSharp/Portable/Declarations/RootSingleNamespaceDeclaration.cs

```diff
@@ -53,6 +57,14 @@ public bool HasAssemblyAttributes
             }
         }
 
+        public override bool HasGlobalUsings
```

Comment:
Does `SingleNamespaceDeclarationEx` need to override this as well? I'm not sure what `SingleNamespaceDeclarationEx` is used for, so just noting about it.

Reply:
> Does SingleNamespaceDeclarationEx need to override this as well? I'm not sure what SingleNamespaceDeclarationEx is used for, so just noting about it.

Namespace declarations are not allowed to have Global Using directives. For the purpose of error recovery all usings there will be treated as non-global, therefore, there are no Global Usings there.

---
In reply to: [582082881](https://github.com/dotnet/roslyn/pull/51426#discussion_r582082881) [](ancestors = 582082881)